storage/innobase/handler/ha_innodb.cc

```diff
@@ -482,6 +482,12 @@ ibool meb_get_checksum_algorithm_enum(const char *algo_name,
 }
 #endif /* !UNIV_HOTBACKUP */
 
+static const char *sys_tablespace_encrypt_names[] = {
+    "OFF", "ON", "RE_ENCRYPTING_TO_KEYRING", NullS};
```

Comment:
IMHO, the name should be simple "TO_KEYRING" or even  "KEYRING" is even better

Reply:
if you are open for suggestions,  please consider these

dte variable values could be shortened to  FROM ONLINE_TO_KEYRING to 'TO_KEYRING' and (ONLINE_FROM_KEYRING_TO_UNENCRYPTED' to "TO_UNENCRYPTED" or "TO_MASTER_KEY"

Reply:
About changing RE_ENCRYPTING_TO_KEYRING - maybe I will try to explain why this name in the first place. We have a system tablespace encrypted with MK encryption, then we decide to encrypt it with encryption threads. You enable encryption threads for it, but I did not want a situation that - you enable encryption threads while having sys_tablespace_encrypt set to ON. You would end up with KEYRING encrypted system tablespace, although sys_tablespace_encrypt would indicated that it is encrypted with MK encryption. Changing it to "TO_KEYRING" or "KEYRING" could indicate that this is enough for system tablespace to be re-encrypted with keyring, although it is not. It is just a marker that encryption threads can re-encrypt sys tablespace.

Reply:
I think this is problem specific to "Forcing MariaDB encryption threads with Oracle/PS".  that is, we have two variables to control encryption of same subsystem.

MariaDB says, just innodb_encrypt_tables=ON with encryption threads is enough to encrypt any tablespace. They dont have other options/variables to control encryption specifically for undo and redo(I assume).  But when it is brought here,

we have two options (dte and innodb_undo_log_encrypt/redo_log_encrypt/ssys_tablespace_encrypt) to control subsystems.

With the fine tuning of encryption options available for these subsystems, may be it is better to control them via their respective options. What do you think?

Reply:
I do not think that fine tuning for all subsystem is needed. Let leave those variables for foreground encryption.